app/js/enhancedDate.js

```diff
@@ -1,5 +1,80 @@
 /* 
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ */
+var enhancedDate = (function enhancedDate() {
 
+    'use strict';
+
+    // Private data, not accessible outside the enhancedDate module
+    var data = {
+        days: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 
+                'Saturday'],
+        months: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 
+                'August', 'September', 'October', 'November', 'December']
+    };
+
+    // check if date has been set, otherwise create a new date object
+    function hasDate() {
+        
+        if (!data.date) {
+            data.date = new Date();
+        }
+    }
+
+    // Methods availabls as a public API
+    return {
+        // setDate allows users to change the date without touching our hidden date variable
+        setDate: function setDate(dateValue) {
+
+            hasDate();
+            if (typeof dateValue === 'number' || dateValue instanceof Date) {
+                data.date = new Date(dateValue);
```

Comment:
perfect - you created your own instance of a Date object rather than setting your internal date to the passed-in Date object, which prevents bugs when the passed-in Date instance is modified externally